file_fetcher.ts

```diff
@@ -328,7 +328,7 @@ export async function fetchWithRetries(
     iterationCount++;
     try {
       const res = await fetch(url, init);
-      if (res.ok || iterationCount > maxRetries) {
+      if (res.ok || iterationCount > maxRetries || res.status === 404) {
```

Comment:
I think all 4xx errors generally fall into that bucket. There are some 5xx errors that probably shouldn't be retried as well.